sdk/provisioning/Azure.Provisioning/src/Utilities/FormattableStringHelpers.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Provisioning.Utilities
+{
+    internal static class FormattableStringHelpers
+    {
+        public static GetPathPartsEnumerator GetFormattableStringFormatParts(ReadOnlySpan<char> format) => new GetPathPartsEnumerator(format);
+
+        public ref struct GetPathPartsEnumerator
+        {
+            private ReadOnlySpan<char> _path;
+            public Part Current { get; private set; }
+
+            public GetPathPartsEnumerator(ReadOnlySpan<char> format)
+            {
+                _path = format;
+                Current = default;
+            }
+
+            public readonly GetPathPartsEnumerator GetEnumerator() => this;
+
+            public bool MoveNext()
+            {
+                var span = _path;
```

Comment:
nit: the name on the ctor is `format`, the field name is `path`, and the local variable is `span` maybe make these consistent?

Reply:
It also appears that you never assign to span, and always just update _path, do we even need this variable?

Reply:
sure, let me refactor this part.
BTW this code might look familiar to you - I copied this from our generator.
Maybe later I will introduce the same change to MTG.